File: C#/problem10.cs
using System;
using System.Diagnostics;

class Problem10
{
  public static void Main(string[] args)
  {
    int max = 2000000;
    if (args.Length > 0)
    {
      max = Convert.ToInt32(args[0]);
    }
    Stopwatch stopwatch = new Stopwatch();
    stopwatch.Start();
    Console.WriteLine(SumOfPrimes(max));
    stopwatch.Stop();
    Console.WriteLine(stopwatch.ElapsedMilliseconds / 1000.0 + " seconds");
  }

  public static long SumOfPrimes(int highestPrime)
  {
    long sum = 0;
    if (highestPrime >= 2)
    {
      sum += 2;
    }
    for (int i = 3; i < highestPrime; i+=2)
    {
      if (Prime(i))
      {
        sum += i;
      }
    }
    return sum;
  }

  public static bool Prime(int n)
  {
    long max = Convert.ToInt64(Math.Ceiling(Math.Sqrt(n)));
    bool prime = true;
    for (long i = 2; i <= max; i++)
    {
      if (n % i == 0)
      {
        prime = false;
        i = max;
      }
    }
    return prime;
  }
}

File: Ruby/problem10.rb
require 'prime'

# Find sum of all primes below input
def sum_of_primes(highest_prime)
  sum = 0
  sum += 2 if highest_prime >= 2
  for i in (3...highest_prime).step(2)
    sum += i if Prime.prime?(i)
  end
  sum
end

max = ARGV.first ? ARGV.first.to_i : 2000000
start_time = Time.now
puts sum_of_primes(max)
puts "#{Time.now - start_time} seconds."
